src/flows/parser_item_html_text_streaming_flow.py

```diff
@@ -41,9 +43,90 @@
 on_error=ABORT_STATEMENT;
 """
 
+# get flow db and schema names from config for ordering tasks
+DATABASE_NAME = ARTICLES_DB_SCHEMA_DICT[ENVIRONMENT]["articles_database_name"]
+SCHEMA_NAME = ARTICLES_DB_SCHEMA_DICT[ENVIRONMENT]["articles_schema_name"]
+
+# create copy of default dict and add flow db and schema names for ordering tasks
+FLOW_SNOWFLAKE_DICT = deepcopy(SNOWFLAKE_DEFAULT_DICT)
+FLOW_SNOWFLAKE_DICT["database"] = DATABASE_NAME
+FLOW_SNOWFLAKE_DICT["schema"] = SCHEMA_NAME
+
+
+# SQL statements to insert new ordered raw data
+# These depend on the initial execution of the 'initial clean' statements in the sql folder
+# This will execute after the copy statement as a single statement
+INSERT_ORDERED_SQL = f"""set max_date = (select max(snowflake_loaded_at) from article_content_ordered_live);
+        
+    insert into article_content_ordered_live (
+        RESOLVED_ID, 
+                HTML, 
+                TEXT, 
+                SNOWFLAKE_LOADED_AT,
+                TEXT_MD5
+    )
+    select RESOLVED_ID, 
+                HTML, 
+                TEXT, 
+                SNOWFLAKE_LOADED_AT,
+                TEXT_MD5
+    from raw.ITEM.ARTICLE_CONTENT_V2
+        where snowflake_loaded_at > $max_date
+        order by resolved_id;"""
+
+# This will run after an hourly update only once on the weekend
+# Will probably need to apply grants to the renamed live table
+# Revoke grants will remove access to the old live table for better UX
+WEEKLY_DEDUP_SQL = f"""
+    -- weekly dedup on 3XL warehouse
+
+    use warehouse dpt_wh_3xl;
+    
+    create or replace table article_content_ordered_new as (
```

Comment:
@braunreyes from what I understand, your plan is to create this final deduped table on a weekly run using a 3XL warehouse. This seems like a good approach by isolating this big operation from DBT by doing it as a Prefect job.

The only question I have is, while you are creating this weekly operation, wouldn't be of advantage to dedupe and merge both the snapshot and the live table into one table?

In the implementation decision you mention: "The final set of tables will be exposed via a dbt view in the analytics database with the ability to further dedup at run time" and the example query seem to suggest doing a _union_ of the snapshot and live tables.
_UNION_: can be an expensive operation for large tables at runtime
Also, the `article_content_ordered_live` is a growing table initialized few months back when we started the process and could get large over time.

Reply:
I have tested this union with the query patterns that Evgeny is using where he is grabbing specific list of 10k rows at a time.  On the medium warehouse he is using it take an average of 30 seconds to return results.  Keeping the snapshot data separate allows me to only have to do a weekly reorder of the raw table.